django/db/models/sql/query.py

```diff
@@ -1054,15 +1054,23 @@ def resolve_lookup_value(self, value, can_reuse, allow_joins, simple_col):
         elif isinstance(value, (list, tuple)):
             # The items of the iterable may be expressions and therefore need
             # to be resolved independently.
+            resolved_values = []
             for sub_value in value:
                 if hasattr(sub_value, 'resolve_expression'):
                     if isinstance(sub_value, F):
-                        sub_value.resolve_expression(
+                        resolved_values.append(sub_value.resolve_expression(
                             self, reuse=can_reuse, allow_joins=allow_joins,
                             simple_col=simple_col,
-                        )
+                        ))
                     else:
-                        sub_value.resolve_expression(self, reuse=can_reuse, allow_joins=allow_joins)
+                        resolved_values.append(sub_value.resolve_expression(
+                            self, reuse=can_reuse, allow_joins=allow_joins
+                        ))
+                else:
+                    resolved_values.append(sub_value)
+            if isinstance(value, tuple):
```

Comment:
Is this necessary? If not I don't see much harm in changing the type to a list internally or even to always convert it to a `tuple` to avoid hashing errors down the line.